drv/Map.drv

```diff
@@ -142,7 +210,459 @@ public interface MAP KEY_VALUE_GENERIC extends FUNCTION KEY_VALUE_GENERIC, Map<K
 
 	@Deprecated
 	@Override
-	boolean containsValue(Object value);
+	default boolean containsValue(Object ov) {
+		if (!(ov instanceof VALUE_CLASS)) return false;
+		return containsValue(VALUE_OBJ2TYPE(ov));
+	}
+
+#endif
+
+	// Defaultable methods
+
+#if KEYS_PRIMITIVE || VALUES_PRIMITIVE
+
+	/** Returns the value to which the specified key is mapped, or the {@code defaultValue} if this map contains no mapping for the key.
+	 *
+	 * @param key the key.
+	 * @param defaultValue the default mapping of the key.
+	 *
+	 * @return the value to which the specified key is mapped, or the {@code defaultValue} if this map contains no mapping for the key.
+	 *
+	 * @see java.util.Map#getOrDefault(Object, Object)
+	 * @since 8.0.0
+	 */
+
+	default VALUE_GENERIC_TYPE GET_VALUE_OR_DEFAULT(KEY_TYPE key, VALUE_GENERIC_TYPE defaultValue) {
+#if VALUES_PRIMITIVE
+		return containsKey(key) ? GET_VALUE(key) : defaultValue;
+#else
+		V v;
+		return (v = get(key)) != null || containsKey(key) ? v : defaultValue;
+#endif
+	}
+
```

Comment:
Difficult question. The point is: why do we have a different code than the primitive case? For me, I would use exactly the same code.

There are some subtle semantics points here. fastutil maps always have settable default return values, which makes the logic "if it returns null or it is mapped to null" logic a bit weird. I mean, why don't we check v != defaultReturnValue()?

So one proposal could be:

final VALUE_GENERIC_TYPE v = get(key), drv = defaultReturnValue();
if (v != drv) return v;
put(key, value);
return drv;

This is possibility that preserves the original semantics in case defaultReturnValue() == null.

The problem is that now the semantics diverges from the primitive case. We have to decide if this is an asset or a liability.